src/main/java/com/gojek/beast/models/ProtoField.java

```diff
@@ -0,0 +1,132 @@
+package com.gojek.beast.models;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.DescriptorProtos;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ProtoField {
+    private String name;
+    private String typeName;
+    private DescriptorProtos.FieldDescriptorProto.Type type;
+    private DescriptorProtos.FieldDescriptorProto.Label label;
+    private List<ProtoField> fields;
+    private int index;
+    private DescriptorProtos.FieldDescriptorProto fieldProto;
+
+    public ProtoField() {
+        this.fields = new ArrayList<>();
+    }
+
+    public ProtoField(DescriptorProtos.FieldDescriptorProto f) {
+        this.fieldProto = f;
+        this.name = f.getName();
+        this.type = f.getType();
+        this.label = f.getLabel();
+        this.index = f.getNumber();
+        this.fields = new ArrayList<>();
+        this.typeName = f.getTypeName();
+    }
+
+    @VisibleForTesting
```

Comment:
I suppose all the VisibleForTesting constructors could be removed and handled in tests